### PR TITLE
Diamond core tests

### PR DESCRIPTION
**Pull request by tsheasha**
As per the standards we hold ourselves to of testing our software, now
we can run tests on the core diamond code for collectors and metrics.
Running the tests for these modules is done with the command:-

`python test.py -d`

This runs the tests for the diamond core *NOT* the collectors. The
collectors are handles in pull/87.

### CODE REVIEW

File: src/diamond/test.py [hunk 1]
```diff
@@ -0,0 +1,327 @@
+#!/usr/bin/env python
+# coding=utf-8
+###############################################################################
+
+import os
```
Comment (zachm) [inline src/diamond/test.py]: import ordering

File: src/diamond/tests/testcollector.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import configobj
+
+from diamond.collector import Collector
+
+
+class BaseCollectorTest(unittest.TestCase):
+
+    @patch('diamond.collector.Collector.publish_metric')
+    def test_SetDimensions(self, mock_publish):
+        """
```
Comment (zachm) [inline src/diamond/tests/testcollector.py]: why is this commented out? if you're going to leave it then say why it's an example or whatever

File: src/diamond/test.py [hunk 3]
```diff
@@ -0,0 +1,327 @@
...
+            f = open(self.getFixturePath(results_name), 'r')
+            data = pickle.load(f)
+            return data
+        finally:
+            f.close()
+
+    def setPickledResults(self, results_name, data):
+        pickle.dump(data, open(self.getFixturePath(results_name), "w+b"))
+
+    def assertUnpublished(self, mock, key, value, expected_value=0):
```
Comment (zachm) [inline src/diamond/test.py]: ??

File: src/diamond/test.py [hunk 4]
```diff
@@ -0,0 +1,327 @@
...
+                      dest="collector",
+                      default="",
+                      help="Run a single collector's unit tests")
+    parser.add_option("-v",
+                      "--verbose",
+                      dest="verbose",
+                      default=1,
+                      action="count",
+                      help="verbose")
+    parser.add_option("-d",
```
Comment (zachm) [inline src/diamond/test.py]: maybe add --diamond-only

Comment (tsheasha) [inline src/diamond/test.py]: sure

File: src/diamond/tests/testcollector.py (hunk 2 above)
Comment (tsheasha) [inline src/diamond/tests/testcollector.py]: I will add this after pull/87 is merged

File: src/diamond/test.py (hunk 1 above)
Comment (tsheasha) [inline src/diamond/test.py]: commented below on the origin of this file

File: src/diamond/test.py (hunk 3 above)
Comment (tsheasha) [inline src/diamond/test.py]: commented below on the origin of this file

**Comment by poros** [comment]
Since we're committed on being as much as possible similar to diamond, we can't do much about code quality
+1

**Pull request merged**